firestore-shorten-urls-dynamic-links/extension.yaml

```diff
@@ -0,0 +1,143 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: firestore-shorten-urls-dynamic-links
+displayName: Shorten URLs with Dynamic Links
+specVersion: v1beta
+version: 0.1.0
+
+description: >
+  Shortens URLs written to a specified Cloud Firestore collection using
+  Firebase Dynamic Links.
+
+license: Apache-2.0
+billingRequired: true
+sourceUrl: https://github.com/firebase/extensions/tree/master/firestore-shorten-urls-dynamic-links
+releaseNotesUrl: https://github.com/firebase/extensions/commits/master
+
+author:
+  authorName: Firebase
+  url: https://firebase.google.com
+
+contributors:
+  - authorName: Kevin Cheung
+    email: kevincheung@google.com
+    url: https://github.com/kevinthecheung
+
+roles:
+  - role: datastore.user
+    reason: Allows the extension to write shortened URLs to Cloud Firestore.
+
+resources:
+  - name: shorten_create
+    type: firebaseextensions.v1beta.function
+    description:
+      Listens for new documents in your specified Cloud Firestore collection,
+      and if they contain URLs, shortens the URLs, then writes the shortened
+      form back to the same document.
+    properties:
+      sourceDirectory: .
+      location: ${LOCATION}
+      runtime: nodejs10
+      eventTrigger:
+        eventType: providers/cloud.firestore/eventTypes/document.create
+        resource: projects/${PROJECT_ID}/databases/(default)/documents/${COLLECTION_PATH}/{documentId}
+  - name: shorten_update
+    type: firebaseextensions.v1beta.function
+    description:
+      Listens for writes of new URLs to your specified Cloud Firestore collection, shortens the URLs,
+      then writes the shortened form back to the same document.
+    properties:
+      sourceDirectory: .
```

Comment:
remove, since your functions are inside "functions" folder, which is the default for "sourceDirectory". Right now, it'll try to look in the root of the extension, which is not where package.json is.